# HG changeset patch
# User Andrew Cooper <andrew.cooper3@citrix.com>
# Date 1380098661 -7200
# Node ID 688f9578f548c65e9901879fc6a52572f3c2bbc9
# Parent  2b3a2e0d2b15316abcbeb83a049366d16a2c054d
ns16550: Use correct #define symbol for HAS_IOPORTS

CID 1091471, Regression caused by 7c1de0038895cbc75ebd0caffc5b0f3f03c5ad51

This appears to be a typo which causes check_existence() to unconditionally
return 1 in all cases.

Signed-off-by: Andrew Cooper <andrew.cooper3@citrix.com>
Acked-by: Ian Campbell <ian.campbell@citrix.com>

diff -r 2b3a2e0d2b15 -r 688f9578f548 xen/drivers/char/ns16550.c
--- a/xen/drivers/char/ns16550.c
+++ b/xen/drivers/char/ns16550.c
@@ -503,7 +503,7 @@ static int __init check_existence(struct
 {
     unsigned char status, scratch, scratch2, scratch3;
 
-#ifdef HAS_IO_PORTS
+#ifdef HAS_IOPORTS
     /*
      * We can't poke MMIO UARTs until they get I/O remapped later. Assume that
      * if we're getting MMIO UARTs, the arch code knows what it's doing.
